backend/tests/unit/core/constraint_validators/test_node_grouped_uniqueness.py

```diff
@@ -75,6 +75,41 @@ async def test_uniqueness_constraint_conflict_two_attribute(
         with pytest.raises(ValidationError, match="Violates uniqueness constraint 'name-color'"):
             await self.__call_system_under_test(db=db, branch=default_branch, node=car_accord_main)
 
+    async def test_uniqueness_constraint_no_conflict_attribute_enum(
+        self,
+        db: InfrahubDatabase,
+        default_branch: Branch,
+        car_accord_main: Node,
+        car_camry_main: Node,
+        car_volt_main: Node,
+    ):
+        car_schema = registry.schema.get("TestCar", branch=default_branch, duplicate=False)
+        car_schema.uniqueness_constraints = [["nbr_seats__value", "name__value"]]
+        attr = car_schema.get_attribute("nbr_seats")
+        attr.optional = False
+        attr.enum = [2, 4, 5, 7]
+
+        await self.__call_system_under_test(db=db, branch=default_branch, node=car_accord_main)
+
+    async def test_uniqueness_constraint_conflict_attribute_enum(
+        self,
+        db: InfrahubDatabase,
+        default_branch: Branch,
+        car_accord_main: Node,
+        car_camry_main: Node,
+        car_volt_main: Node,
+    ):
+        car_schema = registry.schema.get("TestCar", branch=default_branch, duplicate=False)
+        attr = car_schema.get_attribute("nbr_seats")
+        attr.optional = False
+        attr.enum = [2, 4, 5, 7]
+
+        car_accord_main.name.value = "camry"
```

Comment:
I think you can remove this line
and maybe you can set the uniqueness_constraint on the `car_schema` variable above instead of using `car_accord_main.get_schema()`

Reply:
This is what I tried to do in the first place but the test was not picking up the correct schema changes. I expected `car_schema` to point to the same object as `car_accord_main.get_schema()` but it might not be the case I guess?

When having `car_schema.uniqueness_constraints = [["nbr_seats__value", "name__value"]]` instead, running the test actually raises `Violates uniqueness constraint 'name' at name` instead of `Violates uniqueness constraint 'nbr_seats-name'`. Could be my test setup is not correct in the first place?

Reply:
oh I see
`NodeGroupedUniquenessConstraint.check` uses `node.get_schema()` instead of getting the schema from the registry or SchemaBranch

so this test makes sense, disregard my previous comment